CMakeLists.txt

```
@@ -93,9 +96,15 @@ target_link_libraries(rmm INTERFACE CCCL::CCCL)
 target_link_libraries(rmm INTERFACE fmt::fmt-header-only)
 target_link_libraries(rmm INTERFACE spdlog::spdlog_header_only)
 target_link_libraries(rmm INTERFACE dl)
+target_link_libraries(rmm INTERFACE nvtx3-cpp)
 target_compile_features(rmm INTERFACE cxx_std_17 $<BUILD_INTERFACE:cuda_std_17>)
 target_compile_definitions(rmm INTERFACE LIBCUDACXX_ENABLE_EXPERIMENTAL_MEMORY_RESOURCE)
 
+# Disable NVTX if necessary
+if(NOT USE_NVTX)
+  target_compile_definitions(rmm PUBLIC NVTX_DISABLE)
```

Strictly speaking this should probably be INTERFACE.
```suggestion
  target_compile_definitions(rmm INTERFACE NVTX_DISABLE)
```